pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        #
+        if repo.classes:
+            if not self.classes:
+                self.classes = repo.classes
+            else:
+                for ns in repo.classes:
+                    for cl in repo.classes[ns]:
+                        # test for existing qualifier
+                        self.classes[ns][cl.classname] = \
+                            repo.classes[ns][cl.classname]
+        if repo.instances:
+            if repo.instances:
+                if not self.instances:
+                    self.instances = repo.instances
+                else:
+                    for ns in repo.instances:
+                        for inst in repo.instances[ns]:
+                            # TODO test for duplicate instance names.
+                            self.instances[ns].extend(
+                                repo.instances[ns])
+        if repo.qualifiers:
+            if repo.qualifiers:
+                if not self.qualifiers:
+                    self.qualifiers = repo.qualifiers
+                else:
+                    for ns in repo.qualifiers:
+                        for ql in repo.qualifiers[ns]:
+                            # Test for existing qualifier
+                            self.qualifiers[ns][ql.name] = \
+                                repo.qualifiers[ns][ql.name]
+
+    def add_cimobjects(self, objects, namespace=None):
+        """
+            Add  a CIM object (CIMClass, CIMInstance, CIMQualifierDecl)
+            or list of objects to the faked repository.
+
+            This allows the user to create objects directly and insert them
+            into the repository for tests.
+        """
+        if not namespace:
+            namespace = DEFAULT_NAMESPACE
+        if isinstance(objects, list):
+            for obj in objects:
+                self.add_cimobjects(obj, namespace=namespace)
+
+            if self.verbose:
+                self.display_repository()
+
+        else:
+            obj = objects
+            if isinstance(obj, CIMClass):
+                cc = obj
+                if cc.superclass:
+                    try:
+                        _ = self.GetClass(cc.superclass,  # noqa: F841
+                                          namespace=namespace,
+                                          LocalOnly=True,
+                                          IncludeQualifiers=False)
+                    except CIMError as ce:
+                        if ce.args[0] == CIM_ERR_NOT_FOUND:
+                            ce.args = (CIM_ERR_INVALID_SUPERCLASS,
+                                       cc.superclass)
+                            raise
+                        else:
+                            raise
+                try:
+                    # The following generates an exception for each new ns
+                    self.classes[namespace][cc.classname] = cc
+                except KeyError:
+                    self.classes[namespace] = NocaseDict({cc.classname: cc})
```

Comment:
This adds a reference to the CIMClass object provided by the caller. These objects are mutable. This means if the caller modifies that object, the object in the repo is also modified. I don't think we want that. We would need to make a copy.

Similarly for the other types of objects.

Reply:
Done.  We copy everything into the repo.

Reply:
Also copy everything from repo that gets returned with operations